tiledb/sm/misc/utils.cc

```diff
@@ -675,6 +685,32 @@ T safe_mul(T a, T b) {
   return prod;
 }
 
+uint64_t left_p2_m1(uint64_t value) {
+  // Edge case
+  if (value == UINT64_MAX)
+    return value;
+
+  uint64_t ret = 0;  // Min power of 2 minus 1
+  while (ret <= value) {
+    ret = (ret << 1) + 1;  // Next larger power of 2 minus 1
```

Comment:
Hmm I think the algorithm is correct. The change I am making is replacing `+` for `|` as it will probably be faster. But logical operators (or addition with `1`) is faster than subtraction. Please let me know if you see something wrong with correctness, just in case.

Reply:
You are right, my suggestion would be incorrect